### PR TITLE
Add live test

### PR DESCRIPTION
**Pull request by bbenligiray**
Check this out fam, does everything look right?

### CODE REVIEW

**Comment by sassmilic** [comment]
Should this be made configurable (re: gas price selection technique) so that we can easily test different methods (e.g. rand1, const * recommended, etc)?

**Comment by bbenligiray** [comment]
Actually different methods must be tested in parallel, so implemented that. One difference from the real thing is that each method makes a single attempt, doesn't bump gas price every minute. We probably need that too don't we?

File: live-testing/src/live-test.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        .updateData(
+          Math.round(Math.random() * 1000),
+          { gasPrice: usedGasPrice }
+          );
+      // Wait until it's confirmed
+      await receipt.wait();
+      // Find out when it was confirmed
+      const tx = await provider.getTransaction(receipt.hash);
+
+      util.addTx(testCase, {
```
Comment (sassmilic) [inline live-testing/src/live-test.js:90]: I'd like to also log the timestamps (wake up time and confirmation time).

File: live-testing/src/live-test.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        else {
+          usedGasPrice = sampleGasPrice;
+        }
+      }
+      else if (testCase == 'boosted10') {
+        usedGasPrice = recommendedGasPrice.mul(11).div(10);
+      }
+
+      // Make the transaction
+      const receipt = await mockContract
```
Comment (sassmilic) [inline live-testing/src/live-test.js:79]: I'm not too familiar with how `await` works with for-loops. Does the for-loop just keep iterating after the transaction is made, whether or not the transaction is confirmed?

**Review by sassmilic — CHANGES_REQUESTED** [review]
lgtm, some minor comments/questions

**Comment by sassmilic** [comment]
> One difference from the real thing is that each method makes a single attempt, doesn't bump gas price every minute. We probably need that too don't we?
How would the node bump gas up if it's stateless?

File: live-testing/src/live-test.js (hunk 1 above)
Comment (sassmilic) [reply live-testing/src/live-test.js:90]: There are some slight difference when measuring "time" in block numbers and actual timestamps, from my experience.

Comment (bbenligiray) [reply live-testing/src/live-test.js:90]: I assume you're asking for the block timestamps and not the actual time.

File: live-testing/src/live-test.js (hunk 2 above)
Comment (bbenligiray) [reply live-testing/src/live-test.js:79]: Yeah so it's not truly parallel, that's a bit too annoying to implement. One option is to run 4 separate scripts, but we probably don't need that.

**Comment by bbenligiray** [comment]
> How would the node bump gas up if it's stateless?

The node doesn't *see* its previous (pending) fulfillment transaction and makes a new transaction with the updated gas price. If the updated gas price is larger than the previous gas price, the transaction is effectively bumped.

File: live-testing/src/live-test.js (hunk 2 above)
Comment (sassmilic) [reply live-testing/src/live-test.js:79]: Like I said before, you could randomize the order each time to mitigate against some test cases having an advantage.

**Review by sassmilic — APPROVED** [review]

**Pull request merged**